crates/wasmtime/src/runtime/types.rs

```diff
@@ -431,6 +431,132 @@ impl RefType {
 
 /// The heap types that can Wasm can have references to.
 ///
+/// # Subtyping Hierarchy
+///
+/// Wasm has three different heap type hierarchies:
+///
+/// 1. Function types
+/// 2. External types
+/// 3. Internal types
```

Comment:
Out of curiosity is the term "internal" used at the spec level as well? It makes sense to mirror "external" but I figured these'd be called "any types" or... something along those lines. Anyway I'm just musing here.

Reply:
Yeah I've seen it around, eg https://github.com/WebAssembly/gc/blob/main/proposals/gc/MVP.md#heap-types

Also turning an `externref` into an `anyref` is called "internalizing"